internal/pkg/template/diff/formatter.go

```diff
@@ -123,9 +123,26 @@ func (f *keyedFormatter) formatMod(node diffNode) (string, error) {
 	return processMultiline(content, prefixByFn(prefixMod), indentByFn(f.indent)), nil
 }
 
-func (f *keyedFormatter) formatPath(node diffNode) string {
-	content := process(node.key()+":", prefixByFn(prefixMod), indentByFn(f.indent))
-	return content + "\n"
+// joinPath collapses all key names under keyNode into one line as long as there is only modification under the key.
+// For example, if only the `DesiredCount` of an ECS service is changed, then the returned path becomes `/Resources/Service/Properties/DesiredCount`. If multiple entries of an ECS service is changed, then the returned path is: `/Resources/Service/Properties`.
+// It returns the stringified path, as well as the node to which the last key on the path belongs.
+func (f *keyedFormatter) formatPath(keyNode diffNode) (string, diffNode) {
+	content := keyNode.key()
+	for {
+		if len(keyNode.children()) != 1 {
+			break
+		}
+		peek := keyNode.children()[0]
+		if len(peek.children()) == 0 {
+			break
+		}
+		if _, ok := peek.(*node); !ok {
+			break
+		}
+		keyNode = peek
+		content = content + "/" + keyNode.key()
```

Comment:
i'm curious about collapsable keys with a sequence in the path? i.e., only the 2nd policy in an iam role is changed, so the diff could show `MyPolicy/Properties/Policies/1/PolicyDocument/Version` or something